### PR TITLE
AM-526 post form bug

### PR DESCRIPTION
**Pull request by dashton82**

### CODE REVIEW

File: src/SFA.DAS.Campaign.UnitTests/Web/Controllers/EmployerInform/FundingAndApprenticeship/WhenFundingPostCalled.cs [hunk 1]
```diff
@@ -1,30 +1,54 @@
...
+            //Arrange
+            mediator.Setup(x => x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None))
+                .ReturnsAsync(mediatorResult);
+            controller.ModelState.AddModelError("TestError","Error");
+           
+            //Act
+            await controller.Index(model);
+            
+            //Assert
+            mediator.Verify(x=>x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None), Times.Once);
```
Comment (chrisfoster76) [inline src/SFA.DAS.Campaign.UnitTests/Web/Controllers/EmployerInform/FundingAndApprenticeship/WhenFundingPostCalled.cs:51]: Shouldn't the assertion be that the Menu property of the VM is set to the query result or similar?

Super pedantic, but should the test name end _Built_? or _Added_?

File: src/SFA.DAS.Campaign.UnitTests/Web/Controllers/RegisterInterest/WhenUserRegistersInterest.cs [hunk 2]
```diff
@@ -332,6 +334,19 @@ public async Task Then_If_The_Model_Is_Not_Valid_Then_The_User_Data_Collection_S
...
+        public async Task Then_If_The_Model_Is_Not_Valid_Then_The_Menu_Is_Added()
+        {
+            //Arrange
+            _controller.ModelState.AddModelError("FirstName","First name");
+            
+            //Act
+            await _controller.Index(_registerInterestModel);
+
+            //Assert
+            _mediator.Verify(x=>x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None), Times.Once);
```
Comment (chrisfoster76) [inline src/SFA.DAS.Campaign.UnitTests/Web/Controllers/RegisterInterest/WhenUserRegistersInterest.cs:348]: As previous comment, is it sufficient to verify the query was sent, or should the check be on the Menu property of the VM?

**Review by chrisfoster76 — APPROVED** [review]
Some minor comments

File: src/SFA.DAS.Campaign.UnitTests/Web/Controllers/EmployerInform/FundingAndApprenticeship/WhenFundingPostCalled.cs (hunk 1 above)
Comment (dashton82) [reply src/SFA.DAS.Campaign.UnitTests/Web/Controllers/EmployerInform/FundingAndApprenticeship/WhenFundingPostCalled.cs:51]: Have updated the test name and added assertion to check menu on viewmodel

File: src/SFA.DAS.Campaign.UnitTests/Web/Controllers/RegisterInterest/WhenUserRegistersInterest.cs (hunk 2 above)
Comment (dashton82) [reply src/SFA.DAS.Campaign.UnitTests/Web/Controllers/RegisterInterest/WhenUserRegistersInterest.cs:349]: Have added assertion to check view model for menu not being null

**Comment by sony-kotha** [comment]
Testing completed.

**Pull request merged**